### PR TITLE
DAOS-13066 test: Reduce concurrency in NLT

### PR DESCRIPTION
**Pull request by ashleypittman**
Required-githooks: true

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/tests/ftest/cart/util/cart_logtest.py [hunk 1]
```diff
@@ -396,28 +405,23 @@ def _check_pid_from_log_file(self, pid, abort_on_warning, leak_wf, show_memleaks
...
                     if line.fac == 'external':
                         show = False
-                    elif show and server_shutdown and (line.get_msg().endswith(
-                        "DER_SHUTDOWN(-2017): 'Service should shut down'")
-                            or line.get_msg().endswith(
-                                "DER_NOTLEADER(-2008): 'Not service leader'")):
-                        show = False
-                    elif show and line.function == 'rdb_stop':
+                    elif show and server_shutdown and any(map(line.get_msg().endswith,
+                                                              SHUTDOWN_RC)):
```
Comment (daltonbohning) [inline src/tests/ftest/cart/util/cart_logtest.py:418]: Love using `any(map(...))`

File: utils/node_local_test.py [hunk 2]
```diff
@@ -3812,7 +3799,7 @@ def _run_test(ptl=None, function=None, test_cb=None):
             # long-running tests which dominate the time, so whilst a higher value here would
             # work there's no benefit in rushing to finish the quicker tests.  The long-running
             # tests are started first.
-            while len(threads) > 5:
+            while len(threads) > 4:
```
Comment (daltonbohning) [inline utils/node_local_test.py:3802]: Should `4` be a global or class constant?

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/cart/util/cart_logtest.py (hunk 1 above)
Comment (ashleypittman) [reply src/tests/ftest/cart/util/cart_logtest.py:418]: I took this from your suggestion on a related but different PR.

File: utils/node_local_test.py (hunk 2 above)
Comment (ashleypittman) [reply utils/node_local_test.py:3802]: or a command line variable?  I think here is as good as anywhere.

**Comment by ashleypittman** [comment]
This seems to reduce the number of failures we're getting from the NLT test in CI currently.

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**